Add ink and IIIF links to intervention export

kmcelwee opened the pull request.

kmcelwee commented:
- [ ] Rename file to annotation
- [ ] drop intervention type column

kmcelwee commented on derrida/interventions/tests.py:
```diff
@@ -955,6 +955,10 @@ def test_intervention_data(self):
+        assert data['ink'] == annotation.ink
+        assert annotation.canvas.short_id in data['page iiif']
+        assert data['annotation region'] == str(annotation.iiif_image_selection())
+
```
- [ ] Delete interventions.csv and interventions.json from PR

rlskoeser commented on derrida/interventions/management/commands/annotation_data.py:
```diff
@@ -68,13 +67,39 @@ def handle(self, *args, **kwargs):
         # NOTE: using OrderedDict to ensure JSON output follows logical
         # order in Python < 3.6, where dict order is not guaranteed
 
+        if intervention.canvas:
+            canvas_url = reverse('books:canvas-image', kwargs={
+                'slug': intervention.work_instance.slug,
+                'short_id': intervention.canvas.short_id, 
+                'mode': 'large',
+            })
+            iiif_base = ('https://derridas-margins.princeton.edu' + canvas_url).replace('/large/', '/iiif/')
+            page_iiif = iiif_base + 'full/500,/0/default.jpg'
+        else:
+            page_iiif = ''
+
```
More of this is hard-coded and based on string replacement than I'd like (not very semantic/readable).

How about this approach?  (It's cheating a little using "images" as the canvas detail image id so we can generate base url as an "endpoint" that the piffle image client will understand)

```suggestion
      from derrida.common.utils import absolutize_url

      page_iiif = annotation_region = None
     
        if intervention.canvas:
           # get the piffle image api client for the canvas
           img = intervention.canvas.image
           # replace original iiif api endpoint and image id with local proxy iiif url
           img.api_endpoint = absolutize_url(reverse('books:canvas-detail', kwargs={'slug': intervention.work_instance.slug, 'short_id': 'images'})).rstrip('/')
           img.image_id = '%s/iiif' % i.canvas.short_id
           # request 500 width image and convert to str for iiif image url
           page_iiif = str(img.size(width=500))

```

kmcelwee replied:
✅ 49244c4

rlskoeser commented on derrida/interventions/management/commands/annotation_data.py:
```diff
@@ -68,13 +67,39 @@ def handle(self, *args, **kwargs):
+
+        # This assumes that any annotation would have a canvas and iiif_base has
+        #  already been set.
+        if intervention.iiif_image_selection():
+            # `canonicalize` makes a request for each item. While iterating, 
+            #  it may be worth commenting out that method.
+            iiif_query = re.sub(
+                r'https://iiif-cloud.*intermediate_file/', '', 
+                str(intervention.iiif_image_selection().canonicalize())
+            )
+            annotation_region = iiif_base + iiif_query
```
This should be nested under the other `if` for an associated canvas, since it's impossible without an associated canvas and we need those variables.

And the image selection should use the iiif image object constructed above – either by setting the api endpoint & id on the selection version based on the first one, or by copying the region from this one into the first one. Probably better to call canonicalize before changing the endpoint, though!

kmcelwee replied:
✅ 49244c4

rlskoeser commented:
Forgot to say — I don't want the data files committed to the repo! Were they added by accident? Please remove and add to gitignore

kmcelwee commented on derrida/interventions/management/commands/annotation_data.py:
```diff
@@ -68,13 +67,39 @@ def handle(self, *args, **kwargs):
+        # This assumes that any annotation would have a canvas and iiif_base has
+        #  already been set.
+        if intervention.iiif_image_selection():
+            # `canonicalize` makes a request for each item. While iterating, 
```
I'm gonna go with no, but let me know if you change your mind!

kmcelwee commented:
@rlskoeser Your suggestions look a lot cleaner than the regex / string concatenation, thank you! I thought you had said we *wanted* the dataset in the github repo, I got the complete opposite message. I've removed that data and added those files to the gitignore. If you give me the green light, I'm happy to merge and deploy!

rlskoeser commented:
> I thought you had said we _wanted_ the dataset in the github repo, I got the complete opposite message. I've removed that data and added those files to the gitignore.

Sorry for the miscommunication! I wondered about that. I want the datapackage files in the repo, and it makes sense to have a standard path where the data exports will put the files by default and where the datapackage validation can reference them — but I don't actually want the data stored in the repo and versioned with the code.
